### DISCUSSION TITLE
storeToRefs with undefined type not always a Ref

### DISCUSSION THREAD

**Discussion by plehnen** [vuejs/pinia#1349] [Help and Questions] [answered]
### Reproduction

https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAbzgEwKYDNgDtUGUbSoA0cAzgVKgCoQBKGpcAvnOlBCHAORjbACGXANwBYAFDjUAD0iw4MAJ5hUcfPxgqAvInFw4EMDGAQs-ADYB+AFxkYUbAHNRYvSH5Zk66Apvl7WJ3EmZ0kZaHgAYxNyOABXUlQAQViYAAt8QjhtNEwcDMoACi5+FNSuEgRdW3VUGwKAShs1DSyAPjgCypc9fUNjUzMbWI8MbFRkIirXd08KH25y8Sqmesnuh1QYDShSGwQmNb1+CKNovYOg+pCxKKwYkrT8rTiE5MeKVAbr25ikA1OBiQ3B4vFAFMwsrZCDR6OhSAUHukPlc4AB6VFwYGzbw2WEAHj8jnaACMUr0AeZcRgCXZHHAAD5xEa5cbtRnDHJjZBLMRAA

### Steps to reproduce the bug

Create a pinia store with a state property that can be undefined and use storeToRefs to turn it into a ref.

### Expected behavior

Both types should be a Ref.

### Actual behavior

The optional one has type "undefined", but only the Ref's type should be string | undefined.

it is:
`const optional: Ref<string | undefined> | undefined`
it should be:
`const optional: Ref<string | undefined>`

### Additional information

This makes it more complicated to work with such refs, as `optional.value` needs to be checked too, although it always exist.

**Comment by murongg** [comment, marked as answer]
try it ` optional: string | undefined`

**Comment by plehnen** [comment]
Cool. That's working, thanks!

I also had another problem with optional parameters: When you don't define it explicitly as undefined, storeToRefs doesn't work.
So I suggest mentioning this in the docs - or even use TS to force you not to add optional props to the time, might help others. But I am happy to know the solution, so I'll thankfully close this issue.
